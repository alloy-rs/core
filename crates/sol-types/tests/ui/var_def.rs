use alloy_sol_types::sol;

// OK
sol! {
    struct Simple {
        uint a;
    }

    mapping(int => Simple) public simpleMap;
}

// Not OK
sol! {
    struct Complex1 {
        uint[] a;
    }

    mapping(int => Complex1) public complexMap;
}

// OK
sol! {
    struct DoubleComplex {
        Complex2 a;
    }
    struct Complex2 {
        uint[] a;
    }

    mapping(int => DoubleComplex) public complexMap;
}

fn main() {}
