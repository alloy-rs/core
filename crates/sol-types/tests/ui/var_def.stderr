error: invalid state variable type
  --> tests/ui/var_def.rs:18:37
   |
18 |     mapping(int => Complex1) public complexMap;
   |                                     ^^^^^^^^^^
